eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -1,101 +1,136 @@
 # Add 'AzsdkResourceType' member to outputs since actual output types have changed over the years.
 
 function Get-PurgeableGroupResources {
-    param (
-        [Parameter(Mandatory=$true, Position=0)]
-        [string] $ResourceGroupName
-    )
-
-    # Get any Key Vaults that will be deleted so they can be purged later if soft delete is enabled.
-    Get-AzKeyVault @PSBoundParameters | ForEach-Object {
-        # Enumerating vaults from a resource group does not return all properties we required.
-        Get-AzKeyVault -VaultName $_.VaultName | Where-Object { $_.EnableSoftDelete } `
-            | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Key Vault' -PassThru
-    }
+  param (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string] $ResourceGroupName
+  )
+  $purgeableResources = @()
 
-    # Get any Managed HSMs in the resource group, for which soft delete cannot be disabled.
-    Get-AzKeyVaultManagedHsm @PSBoundParameters `
-        | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Managed HSM' -PassThru
-}
+  Write-Verbose "Retrieving deleted Key Vaults from resource group $ResourceGroupName"
+
+  # Get any Key Vaults that will be deleted so they can be purged later if soft delete is enabled.
+  $deletedKeyVaults = Get-AzKeyVault -ResourceGroupName $ResourceGroupName -ErrorAction Ignore | ForEach-Object {
+    # Enumerating vaults from a resource group does not return all properties we required.
+    Get-AzKeyVault -VaultName $_.VaultName | Where-Object { $_.EnableSoftDelete } `
+      | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Key Vault' -PassThru
+  }
+
+  if ($deletedKeyVaults) {
+    Write-Verbose "Found $($deletedKeyVaults.Count) deleted Key Vaults to potentially purge."
+    $purgeableResources += $deletedKeyVaults
+  }
+
+  Write-Verbose "Retrieving deleted Managed HSM Key Vaults from resource group $ResourceGroupName"
+
+  # Get any Managed HSMs in the resource group, for which soft delete cannot be disabled.
+  $deletedHsms = Get-AzKeyVaultManagedHsm -ResourceGroupName $ResourceGroupName -ErrorAction Ignore `
```

Comment:
@heaths do we need to filter this to ones that have EnableSoftDelete as well?

Reply:
No. Managed HSMs always have soft delete enabled, just like new Key Vaults. It cannot be turned off, so checking for `EnableSoftDelete` is redundant. I also double-checked with the service team.